ExpressionEngine/FlowRunnerDependencyExtension.cs

```diff
@@ -25,6 +26,23 @@ public static void AddExpressionEngine(this IServiceCollection services)
             services.AddTransient<IFunction, GreaterFunction>();
         }
 
+        /// <summary>
+        /// Added FunctionDefinition to service collection.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="fromFunctionName">The name of the function, without function parenthesis</param>
+        /// <param name="toExpression">The full expression which is inserted</param>
+        public static void AddFunctionDefinition(this IServiceCollection services, string fromFunctionName, string toExpression)
+        {
+            if (fromFunctionName.EndsWith("()"))
+            {
+                throw new ArgumentError($"{nameof(fromFunctionName)} cannot end in ()");
+            }
+
+            services.AddTransient<IFunctionDefinition, FunctionDefinition>(_ =>
```

Comment:
Is it needed that it creates a new FunctionDefinition every time its used? Why not AddSingleton<IFunctionDefinition>(new FunctionDefition...) ?

The trigger for me was the _ =>, allokating a lamda and new metadata for each parser, where function definitions dont have any state as i see it?

Reply:
Yes, you're right 64de37ba86e9dc6540a5a88bdee63855912778bd